### PR TITLE
Nc news 6

### PR DESCRIPTION
**Pull request by jackmaple2**

### CODE REVIEW

File: __tests__/index.test.js [hunk 1]
```diff
@@ -153,3 +162,70 @@ describe('GET api/articles', () => {
...
 })
 
+describe('GET /api/articles/:article_id/comments', () => {
+    test('GET: 200 repsonds with an array of all comments with all 6 properties required', () => {
+        return request(app)
+        .get('/api/articles/1/comments')
+        .expect(200)
+        .then(({body}) => {
+            const {comments} = body;
+            comments.forEach((comment) => {
```
Comment (Keb84) [inline __tests__/index.test.js:174]: Adding a test to check the length of the array will ensure that you are not passing your test as a false positive. What would happen with your forEach if it was an empty array?

File: __tests__/index.test.js [hunk 2]
```diff
@@ -153,3 +162,70 @@ describe('GET api/articles', () => {
...
+            })
+        })
+    })
+    test('GET: 200 repsonds with an array of all comments for an article_id specified on the endpoint', () => {
+        return request(app)
+        .get('/api/articles/1/comments')
+        .expect(200)
+        .then(({body}) => {
+            const {comments} = body;
+            comments.forEach((comment) => {
```
Comment (Keb84) [inline __tests__/index.test.js:194]: See my comment above ☝️

File: app.js [hunk 3]
```diff
@@ -7,19 +7,30 @@ const {
     getArticles,
     getAllArticles
 } = require('./db/controller/article-controller');
+const {
+    getCommentsByArticleId
+} = require('./db/controller/comments-controller');
 const { getEndpointInformation } = require('./db/controller/endpoints-controller');
 
 app.use(express.json());
```
Comment (Keb84) [inline app.js]: Is this necessary yet? You should only have code that is needed for the request you are making. Do you need express.json for a GET request? 💭

File: db/model/comments-model.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+
+const selectCommentsByArticleId = (article_id) => {
+    return db.query(`SELECT * FROM comments WHERE article_id = $1 
+    ORDER BY created_at desc;`,  [article_id])
+    .then((result) => {
+        if (result.rows.length === 0) {
+            return Promise.reject({
+                status: 404,
+                msg: 'Resource not found'
```
Comment (Keb84) [inline db/model/comments-model.js]: What would happen here if the article_id is in the database but there are no comments yet and is an empty array? Would you want to have a error if the article_id exists and there are no comments yet? You should consider how you would handle this when you are getting the same result for two different outcomes.

File: __tests__/index.test.js [hunk 5]
```diff
@@ -153,3 +162,70 @@ describe('GET api/articles', () => {
...
+    })
+    test('GET: 200 repsonds with an array of all comments for an article_id specified on the endpoint, returned in order with most recent first', () => {
+        return request(app)
+        .get('/api/articles/1/comments')
+        .expect(200)
+        .then(({body}) => {
+            const {comments} = body;
+            expect(comments).toBeSortedBy('created_at', {descending: true});
+        })
+
```
Comment (Keb84) [inline __tests__/index.test.js]: Another good test here should be to check for a status 200 and an empty array of comments 👀

**Review by Keb84 — CHANGES_REQUESTED** [review]
Great work here 🤸 check my comments and resubmit when ready 👍
